internal/scheduler/legacyscheduler.go

```diff
@@ -0,0 +1,619 @@
+	var jobPriority int32
+	podSpec := podSpecFromJob(job)
+	if podSpec != nil && it.SchedulingConfig.Preemption.PriorityClasses != nil {
+		if p, ok := it.SchedulingConfig.Preemption.PriorityClasses[podSpec.PriorityClassName]; ok {
+			jobPriority = p
+		}
+	}
+
+	// Add the resource requests of this job to the total usage for this queue.
+	// We mutate copies of it.roundQueueResources and it.totalQueueResources.
+	// Later, if the job is scheduled, we update it.round... and it.total in-place.
+	//
+	// TODO: Account for resource usage separately by priority.
+	jobTotalResourceRequests := common.TotalJobResourceRequest(job)
+	roundQueueResources := it.roundQueueResources.DeepCopy()
+	totalQueueResources := it.totalQueueResources.DeepCopy()
+	totalQueueResourcesByPriority := it.totalQueueResourcesByPriority.DeepCopy()
+	jobSchedulingReport.RoundQueueResources = roundQueueResources
+	jobSchedulingReport.TotalQueueResources = totalQueueResources
+	for resourceType, quantity := range jobTotalResourceRequests {
+		q := totalQueueResources.Resources[resourceType]
+		q.Add(quantity)
+		totalQueueResources.Resources[resourceType] = q
+
+		q = roundQueueResources.Resources[resourceType]
+		q.Add(quantity)
+		roundQueueResources.Resources[resourceType] = q
+
+		rl := totalQueueResourcesByPriority[jobPriority]
+		if rl.Resources == nil {
+			rl.Resources = make(map[string]resource.Quantity)
+		}
+		q = rl.Resources[resourceType]
+		q.Add(quantity)
+		rl.Resources[resourceType] = q
+		totalQueueResourcesByPriority[jobPriority] = rl
+	}
+
+	// Check that the job is large enough for this executor.
+	if ok, reason := it.jobIsLargeEnough(jobTotalResourceRequests); !ok {
+		jobSchedulingReport.UnschedulableReason = reason
+		return jobSchedulingReport, nil
+	}
+
+	// Check total per-queue resource limits.
+	if exceeded, reason := it.exceedsResourceLimits(
+		ctx,
+		totalQueueResources,
+		it.SchedulingConfig.MaximalResourceFractionPerQueue,
+	); exceeded {
+		jobSchedulingReport.UnschedulableReason = reason + " (total limit for this queue)"
+		return jobSchedulingReport, nil
+	}
+
+	// Check per-round resource limits for this queue.
+	if exceeded, reason := it.exceedsResourceLimits(
+		ctx,
+		roundQueueResources,
+		it.SchedulingConfig.MaximalResourceFractionToSchedulePerQueue,
+	); exceeded {
+		jobSchedulingReport.UnschedulableReason = reason + " (per scheduling round limit for this queue)"
+		return jobSchedulingReport, nil
+	}
+
+	// If a NodeDb is provided, try to find a node on which this job can be scheduled.
+	if it.NodeDb != nil {
+		podReport, err := it.selectNodeForPod(ctx, jobId, job)
+		if err != nil {
+			jobSchedulingReport.UnschedulableReason = err.Error()
+			return nil, err
+		}
+		jobSchedulingReport.PodSchedulingReports = append(jobSchedulingReport.PodSchedulingReports, podReport)
+		if podReport.Node == nil {
+			jobSchedulingReport.UnschedulableReason = "pod does not fit on any node"
```

Comment:
will it be obvious why it doesn't fit?  If not we don't need to fix this now, just an idea for the future.

Reply:
Yes, a more detailed reason is stored in the `PodSchedulingReport` embedded in the `JobSchedulingReport`.